### ISSUE TITLE
Failed to access ‘*.jsd.qc’ (ENCSR936XTK test sample)

### ISSUE THREAD

**Issue by dcdxy** [ENCODE-DCC/chip-seq-pipeline2#22]
When attempting to run AQUAS on the test sample datasets (ENCSR936XTK FASTQs), I receive:

`[2018-09-13 15:32:39,95] [error] WorkflowManagerActor Workflow 0846c6f1-989d-4622-8512-b1aebfb5ea97 failed (during ExecutingWorkflowState): Job chip.fingerprint:NA:1 exited with return code 1 which has not been declared as a valid return code.`

which seems to arise from

`ln: failed to access ‘*.jsd.qc’: No such file or directory`.

Any suggestions would be greatly appreciated. Thank you!

**Details:**
I installed AQUAS on a SGE cluster.
**Cromwell:** cromwell-34
**Conda:** conda 4.3.30

[debug_22.tar.gz](https://github.com/ENCODE-DCC/chip-seq-pipeline2/files/2384060/debug_22.tar.gz)

**Comment by leepc12** [comment]
Did you activate your Conda environment (`source activate encode-chip-seq-pipeline`) before running a pipeline? If so please do the following for debugging:

```
$ source activate encode-chip-seq-pipeline
$ which plotFingerprint
$ plotFingerprint --help
```

Also, check your SGE configuration.
```
$ qconf -sconf
$ qconf -spl
```

**Comment by dcdxy** [comment]
Yes, I had activated the Conda environment. 
```
$ which plotFingerprint
~/encode-chip-seq-pipeline/bin/plotFingerprint

$ plotFingerprint --help
usage: An example usage is: plotFingerprint -b treatment.bam control.bam -plot fingerprint.png

This tool samples indexed BAM files and plots a profile of cumulative read
coverages for each. All reads overlapping a window (bin) of the specified
length are counted; these counts are sorted and the cumulative sum is finally
plotted.

Required arguments: [...]
```
SGE configuration:

```
$ qconf -sconf
#global:
execd_spool_dir              /sge/middle_earth/spool
mailer                       /bin/mail
xterm                        /usr/bin/xterm
load_sensor                  none
prolog                       none
epilog                       none
shell_start_mode             posix_compliant
login_shells                 sh,bash,ksh,csh,tcsh
min_uid                      0
min_gid                      0
user_lists                   none
xuser_lists                  none
projects                     none
xprojects                    none
enforce_project              false
enforce_user                 auto
load_report_time             00:00:40
max_unheard                  00:05:00
reschedule_unknown           02:00:00
loglevel                     log_warning
administrator_mail           none
set_token_cmd                none
pag_cmd                      none
token_extend_time            none
shepherd_cmd                 none
qmaster_params               none
execd_params                 none
reporting_params             accounting=true reporting=false \
                             flush_time=00:00:15 joblog=false sharelog=00:00:00
finished_jobs                100
gid_range                    20000-20100
qlogin_command               builtin
qlogin_daemon                builtin
rlogin_command               builtin
rlogin_daemon                builtin
rsh_command                  builtin
rsh_daemon                   builtin
max_aj_instances             2000
max_aj_tasks                 5000
max_u_jobs                   5000
max_jobs                     0
max_advance_reservations     0
auto_user_oticket            0
auto_user_fshare             100
auto_user_default_project    none
auto_user_delete_time        86400
delegated_file_staging       false
reprioritize                 false
jsv_url                      none
libjvm_path                  /etc/alternatives/jre/lib/amd64/server/libjvm.so
additional_jvm_args          -Xmx256m
jsv_allowed_mod              ac,h,i,e,o,j,M,N,p,w

$ qconf -spl
make
ompi
smp
```
I set `"sge_pe" : "smp"` in sge.json.

**Comment by leepc12** [comment]
It's weird `~/encode-chip-seq-pipeline/bin/plotFingerprint`. Please run the following commands for more info.

```
$ which conda
$ conda env list
$ source activate encode-chip-seq-pipeline
$ which deeptools
$ plotFingerprint --version
$ deeptools --version
$ echo $PATH
```

Also, I would like to see full output of `plotFingerprint --help`.

**Comment by dcdxy** [comment]
Oops that should have been `~/miniconda3/envs/encode-chip-seq-pipeline/bin/plotFingerprint`—sorry.

Sure, here are the outputs (I replaced home directory paths with $HOME):

```
$ which conda
~/miniconda3/bin/conda

$ conda env list
# conda environments:
#
aquas_chipseq            $HOME/miniconda3/envs/aquas_chipseq
aquas_chipseq_py3        $HOME/miniconda3/envs/aquas_chipseq_py3
encode-chip-seq-pipeline     $HOME/miniconda3/envs/encode-chip-seq-pipeline
encode-chip-seq-pipeline-python3     $HOME/miniconda3/envs/encode-chip-seq-pipeline-python3
root                   *  $HOME/miniconda3

$ which deeptools
~/miniconda3/envs/encode-chip-seq-pipeline/bin/deeptools

$ which plotFingerprint
~/miniconda3/envs/encode-chip-seq-pipeline/bin/plotFingerprint

$ plotFingerprint --version
plotFingerprint 2.5.4

$ deeptools --version
deeptools 2.5.4

$ echo $PATH
$HOME/miniconda3/envs/encode-chip-seq-pipeline/bin:$HOME/miniconda3/bin:/mnt/work1/software/meme/4.11.4/bin:/mnt/work1/software/openmpi/1.8.1/bin:/mnt/work1/software/perl/5.18.1/bin:/mnt/work1/software/python/2.7/bin:$HOME/bin/jre-10.0.2/bin:$HOME/bin:$HOME/.local/bin:$HOME/local/bin:$HOME/.bds:$HOME/perl5/bin:/sge/bin/linux-x64:/usr/lib64/qt-3.3/bin:/mnt/work1/software/python/3.4.3/bin:/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/sbin:/mnt/work1/software/crossbow-1.2.0:/mnt/work1/software/crossbow-1.2.0/bin/linux64:/hadoop/hadoop-1.1.1/bin:/mnt/work1/software/SNVMix2-0.11.8-r5:/opt/dell/srvadmin/bin:$HOME/bin

$ plotFingerprint --help
usage: An example usage is: plotFingerprint -b treatment.bam control.bam -plot fingerprint.png

This tool samples indexed BAM files and plots a profile of cumulative read
coverages for each. All reads overlapping a window (bin) of the specified
length are counted; these counts are sorted and the cumulative sum is finally
plotted.

Required arguments:
  --bamfiles bam files [bam files ...], -b bam files [bam files ...]
                        List of indexed BAM files (default: None)

Output:
  --plotFile , -plot    File name of the output figure. The file ending will
                        be used to determine the image format. The available
                        options are typically: "png", "eps", "pdf" and "svg",
                        e.g. : fingerprint.png. (default: None)
  --outRawCounts        Output file name to save the read counts per bin.
                        (default: None)

Read processing options:
  --extendReads [INT bp], -e [INT bp]
                        This parameter allows the extension of reads to
                        fragment size. If set, each read is extended, without
                        exception. *NOTE*: This feature is generally NOT
                        recommended for spliced-read data, such as RNA-seq, as
                        it would extend reads over skipped regions. *Single-
                        end*: Requires a user specified value for the final
                        fragment length. Reads that already exceed this
                        fragment length will not be extended. *Paired-end*:
                        Reads with mates are always extended to match the
                        fragment size defined by the two read mates. Unmated
                        reads, mate reads that map too far apart (>4x fragment
                        length) or even map to different chromosomes are
                        treated like single-end reads. The input of a fragment
                        length value is optional. If no value is specified, it
                        is estimated from the data (mean of the fragment size
                        of all mate reads). (default: False)
  --ignoreDuplicates    If set, reads that have the same orientation and start
                        position will be considered only once. If reads are
                        paired, the mate's position also has to coincide to
                        ignore a read. (default: False)
  --minMappingQuality INT
                        If set, only reads that have a mapping quality score
                        of at least this are considered. (default: None)
  --centerReads         By adding this option, reads are centered with respect
                        to the fragment length. For paired-end data, the read
                        is centered at the fragment length defined by the two
                        ends of the fragment. For single-end data, the given
                        fragment length is used. This option is useful to get
                        a sharper signal around enriched regions. (default:
                        False)
  --samFlagInclude INT  Include reads based on the SAM flag. For example, to
                        get only reads that are the first mate, use a flag of
                        64. This is useful to count properly paired reads only
                        once, as otherwise the second mate will be also
                        considered for the coverage. (default: None)
  --samFlagExclude INT  Exclude reads based on the SAM flag. For example, to
                        get only reads that map to the forward strand, use
                        --samFlagExclude 16, where 16 is the SAM flag for
                        reads that map to the reverse strand. (default: None)
  --minFragmentLength INT
                        The minimum fragment length needed for read/pair
                        inclusion. This option is primarily useful in ATACseq
                        experiments, for filtering mono- or di-nucleosome
                        fragments. (default: 0)
  --maxFragmentLength INT
                        The maximum fragment length needed for read/pair
                        inclusion. (default: 0)

Optional arguments:
  --help, -h            show this help message and exit
  --labels  [ ...], -l  [ ...]
                        List of labels to use in the output. If not given, the
                        file names will be used instead. Separate the labels
                        by spaces. (default: None)
  --binSize BINSIZE, -bs BINSIZE
                        Window size in base pairs to sample the genome.
                        (default: 500)
  --numberOfSamples NUMBEROFSAMPLES, -n NUMBEROFSAMPLES
                        Number of bins that sampled from the genome, for which
                        the overlapping number of reads is computed. (default:
                        500000.0)
  --plotFileFormat      image format type. If given, this option overrides the
                        image format based on the ending given via --plotFile
                        ending. The available options are: "png", "eps", "pdf"
                        and "svg" (default: None)
  --plotTitle PLOTTITLE, -T PLOTTITLE
                        Title of the plot, to be printed on top of the
                        generated image. Leave blank for no title. (default: )
  --skipZeros           If set, then regions with zero overlapping readsfor
                        *all* given BAM files are ignored. This will result in
                        a reduced number of read counts than that specified in
                        --numberOfSamples (default: False)
  --outQualityMetrics FILE.txt
                        Quality metrics can optionally be output to this file.
                        The file will have one row per input BAM file and
                        columns containing a number of metrics. Please see the
                        online documentation for a longer explanation: http://
                        deeptools.readthedocs.io/en/latest/content/feature/plo
                        tFingerprint_QC_metrics.html . (default: None)
  --JSDsample sample.bam
                        Reference sample against which to compute the Jensen-
                        Shannon distance and the CHANCE statistics. If this is
                        not specified, then these will not be calculated. If
                        --outQualityMetrics is not specified then this will be
                        ignored. The Jensen-Shannon implementation is based on
                        code from Sitanshu Gakkhar at BCGSC. The CHANCE
                        implementation is based on code from Matthias Haimel.
                        (default: None)
  --version             show program's version number and exit
  --region CHR:START:END, -r CHR:START:END
                        Region of the genome to limit the operation to - this
                        is useful when testing parameters to reduce the
                        computing time. The format is chr:start:end, for
                        example --region chr10 or --region
                        chr10:456700:891000. (default: None)
  --blackListFileName BED file [BED file ...], -bl BED file [BED file ...]
                        A BED or GTF file containing regions that should be
                        excluded from all analyses. Currently this works by
                        rejecting genomic chunks that happen to overlap an
                        entry. Consequently, for BAM files, if a read
                        partially overlaps a blacklisted region or a fragment
                        spans over it, then the read/fragment might still be
                        considered. Please note that you should adjust the
                        effective genome size, if relevant. (default: None)
  --numberOfProcessors INT, -p INT
                        Number of processors to use. Type "max/2" to use half
                        the maximum number of processors or "max" to use all
                        available processors. (default: max/2)
  --verbose, -v         Set to see processing messages. (default: False)
```

**Comment by dcdxy** [comment]
Hi Jin, 
Just wondering if you had any other suggestions concerning this issue?
I appreciate your help!

**Comment by leepc12** [comment]
@shanmukhasampath Please make a new issue for this. This doesn't look like the same problem.

**Comment by leepc12** [comment]
@dcdxy : can you update the pipeline and use singularity backend instead of Conda to avoid these annoying Conda dependency errors?

I have added an instruction for SGE + singularity https://github.com/ENCODE-DCC/atac-seq-pipeline/blob/master/docs/tutorial_sge.md#for-singularity-users

You may need to install [singularity](https://singularity.lbl.gov/) first.

**Comment by leepc12** [comment]
@shanmukhasampath please switch to the SGE + singularity backend. 

https://github.com/ENCODE-DCC/atac-seq-pipeline/blob/master/docs/tutorial_sge.md#for-singularity-users

**Comment by dcdxy** [comment]
Unfortunately, our server is too old to install newer versions of Singularity (>=2.5.2). Is there another way around this issue using Conda?

**Comment by leepc12** [comment]
Please add `"chip.no_jsd" : true,` to your input JSON file until this issue is fixed. This will disable the erroneous fingerprint step.

**Comment by dcdxy** [comment]
After adding this line to `ENCSR936XTK_subsampled.json` and running again, I receive:

```
[2018-09-25 13:31:24,47] [error] WorkflowManagerActor Workflow 1e836437-9e03-
48c9-b975-b9a231a2c886 failed (during ExecutingWorkflowState): Job chip.xcor:1:1 
exited with return code 1 which has not been declared as a valid return code. See
'continueOnReturnCode' runtime attribute for more details.
```
arising from
```
subprocess.CalledProcessError: Command 'Rscript --max-ppsize=500000 $(which run_spp.R) 
-rf -c=rep2-R1.subsampled.67.merged.trim_50bp.no_chrM.15.0M.tagAlign.gz -p=2 
-filtchr=chrM -savp=rep2-R1.subsampled.67.merged.trim_50bp.no_chrM.15.0M.cc.plot.pdf 
-out=rep2-R1.subsampled.67.merged.trim_50bp.no_chrM.15.0M.cc.qc ' returned non-zero 
exit status 1
```
Another exception occurs as well:
```
ProcessLookupError: [Errno 3] No such process
ln: failed to access ‘*.cc.plot.pdf’: No such file or directory
ln: failed to access ‘*.cc.plot.png’: No such file or directory
ln: failed to access ‘*.cc.qc’: No such file or directory
ln: failed to access ‘*.cc.fraglen.txt’: No such file or directory
```

**Comment by leepc12** [comment]
Did you activate the Conda env before running a pipeline? Please upload a tar ball with all logs.

**Comment by dcdxy** [comment]
Yes, I had activated the Conda environment. Please find that tarball [above](https://github.com/ENCODE-DCC/chip-seq-pipeline2/issues/22#issuecomment-424439840).

**Comment by leepc12** [comment]
Can you run the following command on your working directory?
```
FILE_PATH=$(find -name rep2-R1.subsampled.67.merged.trim_50bp.no_chrM.15.0M.tagAlign.gz)
zcat ${FILE_PATH} | wc -l
zcat ${FILE_PATH} | head
```

**Comment by leepc12** [comment]
Please run this for debugging:
```
$ find -name rep2-R1.subsampled.67.merged.* | xargs ls -l
```

**Comment by leepc12** [comment]
Please run the following:
```
$ find -name *.fastq.gz | xargs ls -l
```

**Comment by dcdxy** [comment]
Please find the output below:

```
$ find -name *.fastq.gz | xargs ls -l
-rw-r--r-- 15 dcdxy hgroup 48102097 Feb  1  2018 ./call-bwa_ctl/shard-0/inputs/-1759515581/merge_fastqs_R1_ctl1-R1.subsampled.80.merged.fastq.gz                                                                   
-rw-r--r-- 15 dcdxy hgroup 48737290 Feb  1  2018 ./call-bwa_ctl/shard-0/inputs/-1759515581/merge_fastqs_R2_ctl1-R2.subsampled.80.merged.fastq.gz                                                                   
-rw-r--r-- 15 dcdxy hgroup 46012398 Feb  1  2018 ./call-bwa_ctl/shard-1/inputs/-1767274940/merge_fastqs_R1_ctl2-R1.subsampled.80.merged.fastq.gz                                                                   
-rw-r--r-- 15 dcdxy hgroup 46598569 Feb  1  2018 ./call-bwa_ctl/shard-1/inputs/-1767274940/merge_fastqs_R2_ctl2-R2.subsampled.80.merged.fastq.gz                                                                   
-rw-r--r--  3 dcdxy hgroup       20 Sep 25 12:53 ./call-bwa_R1/shard-0/inputs/-1170548376/merge_fastqs_R1_rep1-R1.subsampled.67.merged.trim_50bp.fastq.gz                                                          
-rw-r--r--  3 dcdxy hgroup       20 Sep 25 12:53 ./call-bwa_R1/shard-1/inputs/-1178307735/merge_fastqs_R1_rep2-R1.subsampled.67.merged.trim_50bp.fastq.gz                                                          
-rw-r--r-- 18 dcdxy hgroup 40893958 Jan 30  2018 ./call-bwa/shard-0/inputs/-739139993/merge_fastqs_R1_rep1-R1.subsampled.67.merged.fastq.gz                                                                        
-rw-r--r-- 15 dcdxy hgroup 42644527 Jan 30  2018 ./call-bwa/shard-0/inputs/-739139993/merge_fastqs_R2_rep1-R2.subsampled.67.merged.fastq.gz                                                                        
-rw-r--r-- 18 dcdxy hgroup 50817632 Jan 30  2018 ./call-bwa/shard-1/inputs/-746899352/merge_fastqs_R1_rep2-R1.subsampled.67.merged.fastq.gz                                                                        
-rw-r--r-- 15 dcdxy hgroup 52071416 Jan 30  2018 ./call-bwa/shard-1/inputs/-746899352/merge_fastqs_R2_rep2-R2.subsampled.67.merged.fastq.gz                                                                        
-rw-r--r-- 15 dcdxy hgroup 48102097 Feb  1  2018 ./call-merge_fastq_ctl/shard-0/execution/glob-4f26c666d13d1cb48973da7f646a7de2/merge_fastqs_R1_ctl1-R1.subsampled.80.merged.fastq.gz                              
-rw-r--r-- 15 dcdxy hgroup 48737290 Feb  1  2018 ./call-merge_fastq_ctl/shard-0/execution/glob-4f26c666d13d1cb48973da7f646a7de2/merge_fastqs_R2_ctl1-R2.subsampled.80.merged.fastq.gz                              
-rw-r--r-- 15 dcdxy hgroup 48102097 Feb  1  2018 ./call-merge_fastq_ctl/shard-0/execution/merge_fastqs_R1_ctl1-R1.subsampled.80.merged.fastq.gz                                                                    
-rw-r--r-- 15 dcdxy hgroup 48737290 Feb  1  2018 ./call-merge_fastq_ctl/shard-0/execution/merge_fastqs_R2_ctl1-R2.subsampled.80.merged.fastq.gz                                                                    
-rw-r--r-- 15 dcdxy hgroup 48102097 Feb  1  2018 ./call-merge_fastq_ctl/shard-0/inputs/-880878542/ctl1-R1.subsampled.80.fastq.gz                                                                                   
-rw-r--r-- 15 dcdxy hgroup 48737290 Feb  1  2018 ./call-merge_fastq_ctl/shard-0/inputs/-880878542/ctl1-R2.subsampled.80.fastq.gz                                                                                   
-rw-r--r-- 15 dcdxy hgroup 46012398 Feb  1  2018 ./call-merge_fastq_ctl/shard-1/execution/glob-4f26c666d13d1cb48973da7f646a7de2/merge_fastqs_R1_ctl2-R1.subsampled.80.merged.fastq.gz                              
-rw-r--r-- 15 dcdxy hgroup 46598569 Feb  1  2018 ./call-merge_fastq_ctl/shard-1/execution/glob-4f26c666d13d1cb48973da7f646a7de2/merge_fastqs_R2_ctl2-R2.subsampled.80.merged.fastq.gz                              
-rw-r--r-- 15 dcdxy hgroup 46012398 Feb  1  2018 ./call-merge_fastq_ctl/shard-1/execution/merge_fastqs_R1_ctl2-R1.subsampled.80.merged.fastq.gz                                                                    
-rw-r--r-- 15 dcdxy hgroup 46598569 Feb  1  2018 ./call-merge_fastq_ctl/shard-1/execution/merge_fastqs_R2_ctl2-R2.subsampled.80.merged.fastq.gz                                                                    
-rw-r--r-- 15 dcdxy hgroup 46012398 Feb  1  2018 ./call-merge_fastq_ctl/shard-1/inputs/-880878542/ctl2-R1.subsampled.80.fastq.gz                                                                                   
-rw-r--r-- 15 dcdxy hgroup 46598569 Feb  1  2018 ./call-merge_fastq_ctl/shard-1/inputs/-880878542/ctl2-R2.subsampled.80.fastq.gz                                                                                   
-rw-r--r-- 18 dcdxy hgroup 40893958 Jan 30  2018 ./call-merge_fastq/shard-0/execution/glob-4f26c666d13d1cb48973da7f646a7de2/merge_fastqs_R1_rep1-R1.subsampled.67.merged.fastq.gz                                  
-rw-r--r-- 15 dcdxy hgroup 42644527 Jan 30  2018 ./call-merge_fastq/shard-0/execution/glob-4f26c666d13d1cb48973da7f646a7de2/merge_fastqs_R2_rep1-R2.subsampled.67.merged.fastq.gz                                  
-rw-r--r-- 18 dcdxy hgroup 40893958 Jan 30  2018 ./call-merge_fastq/shard-0/execution/merge_fastqs_R1_rep1-R1.subsampled.67.merged.fastq.gz                                                                        
-rw-r--r-- 15 dcdxy hgroup 42644527 Jan 30  2018 ./call-merge_fastq/shard-0/execution/merge_fastqs_R2_rep1-R2.subsampled.67.merged.fastq.gz                                                                        
-rw-r--r-- 18 dcdxy hgroup 40893958 Jan 30  2018 ./call-merge_fastq/shard-0/inputs/-880878542/rep1-R1.subsampled.67.fastq.gz                                                                                       
-rw-r--r-- 15 dcdxy hgroup 42644527 Jan 30  2018 ./call-merge_fastq/shard-0/inputs/-880878542/rep1-R2.subsampled.67.fastq.gz                                                                                       
-rw-r--r-- 18 dcdxy hgroup 50817632 Jan 30  2018 ./call-merge_fastq/shard-1/execution/glob-4f26c666d13d1cb48973da7f646a7de2/merge_fastqs_R1_rep2-R1.subsampled.67.merged.fastq.gz                                  
-rw-r--r-- 15 dcdxy hgroup 52071416 Jan 30  2018 ./call-merge_fastq/shard-1/execution/glob-4f26c666d13d1cb48973da7f646a7de2/merge_fastqs_R2_rep2-R2.subsampled.67.merged.fastq.gz                                  
-rw-r--r-- 18 dcdxy hgroup 50817632 Jan 30  2018 ./call-merge_fastq/shard-1/execution/merge_fastqs_R1_rep2-R1.subsampled.67.merged.fastq.gz                                                                        
-rw-r--r-- 15 dcdxy hgroup 52071416 Jan 30  2018 ./call-merge_fastq/shard-1/execution/merge_fastqs_R2_rep2-R2.subsampled.67.merged.fastq.gz                                                                        
-rw-r--r-- 18 dcdxy hgroup 50817632 Jan 30  2018 ./call-merge_fastq/shard-1/inputs/-880878542/rep2-R1.subsampled.67.fastq.gz                                                                                       
-rw-r--r-- 15 dcdxy hgroup 52071416 Jan 30  2018 ./call-merge_fastq/shard-1/inputs/-880878542/rep2-R2.subsampled.67.fastq.gz                                                                                       
-rw-r--r--  3 dcdxy hgroup       20 Sep 25 12:53 ./call-trim_fastq/shard-0/execution/glob-61db8dea51a2a3b9b418bc9d1b7070d5/merge_fastqs_R1_rep1-R1.subsampled.67.merged.trim_50bp.fastq.gz                         
-rw-r--r--  3 dcdxy hgroup       20 Sep 25 12:53 ./call-trim_fastq/shard-0/execution/merge_fastqs_R1_rep1-R1.subsampled.67.merged.trim_50bp.fastq.gz                                                               
-rw-r--r-- 18 dcdxy hgroup 40893958 Jan 30  2018 ./call-trim_fastq/shard-0/inputs/-739139993/merge_fastqs_R1_rep1-R1.subsampled.67.merged.fastq.gz                                                                 
-rw-r--r--  3 dcdxy hgroup       20 Sep 25 12:53 ./call-trim_fastq/shard-1/execution/glob-61db8dea51a2a3b9b418bc9d1b7070d5/merge_fastqs_R1_rep2-R1.subsampled.67.merged.trim_50bp.fastq.gz                         
-rw-r--r--  3 dcdxy hgroup       20 Sep 25 12:53 ./call-trim_fastq/shard-1/execution/merge_fastqs_R1_rep2-R1.subsampled.67.merged.trim_50bp.fastq.gz                                                               
-rw-r--r-- 18 dcdxy hgroup 50817632 Jan 30  2018 ./call-trim_fastq/shard-1/inputs/-746899352/merge_fastqs_R1_rep2-R1.subsampled.67.merged.fastq.gz
```

**Comment by leepc12** [comment]
We figured this out and will make a hot fix this week. Thanks for reporting this.

**Comment by dcdxy** [comment]
I was able to run the updated pipeline successfully on a few of my datasets, but the same issue has since resurfaced for a different dataset:

```
[2018-11-12 18:56:02,86] [error] WorkflowManagerActor Workflow bdcbb718-b33f-44f2-\
b51f-f8fc0f53712b failed (during ExecutingWorkflowState): Job chip.fingerprint:NA:1\
exited with return code 1 which has not been declared as a valid return code. See \
'continueOnReturnCode' runtime attribute for more details.

[...]

ln: failed to access ‘*.jsd.qc’: No such file or directory
```

I had activated the `encode-chip-seq-pipeline` Conda environment. I've attached the new log files.
[debug_22_3.tar.gz](https://github.com/ENCODE-DCC/chip-seq-pipeline2/files/2576645/debug_22_3.tar.gz)

**Comment by leepc12** [comment]
This has already been fixed in this commit (https://github.com/ENCODE-DCC/chip-seq-pipeline2/pull/35/commits/5bd3b7b083630383dc8d71c09fb19c71fa137b6e) and will be included in the next release (v1.1.1).

But this error means that your dataset has a trivially low number of reads?
```
$ cat ./call-bwa/shard-0/execution/SRR4413836-OCT4-WT-rep1_1.merged.flagstat.qc
2 + 0 in total (QC-passed reads + QC-failed reads)
0 + 0 secondary
0 + 0 supplementary
0 + 0 duplicates
1 + 0 mapped (50.00%:-nan%)
2 + 0 paired in sequencing
1 + 0 read1
1 + 0 read2
0 + 0 properly paired (0.00%:-nan%)
0 + 0 with itself and mate mapped
1 + 0 singletons (50.00%:-nan%)
0 + 0 with mate mapped to a different chr
0 + 0 with mate mapped to a different chr (mapQ>=5)
```

**Comment by dcdxy** [comment]
Ok, I see. I made a mistake in assigning each end file (i.e. `*rep1_R1` and `*rep1_R2`) to what were actually separate sequencing runs. Thanks!

**Comment by dcdxy** [comment]
I receive again the same error (comments on September 25) on a different dataset:

```
[2018-11-26 11:13:35,00] [error] WorkflowManagerActor Workflow 11f011ae-0014-470d-
9fc7-cb896683bf60 failed (during ExecutingWorkflowState): Job chip.xcor:2:1 exited 
with return code 1 which has not been declared as a valid return code. See 
'continueOnReturnCode' runtime attribute for more details.
[...]
ln: failed to access ‘*.cc.plot.pdf’: No such file or directory
ln: failed to access ‘*.cc.plot.png’: No such file or directory
ln: failed to access ‘*.cc.qc’: No such file or directory
ln: failed to access ‘*.cc.fraglen.txt’: No such file or directory
```
Please find the tarball below.
[debug_22_4.tar.gz](https://github.com/ENCODE-DCC/chip-seq-pipeline2/files/2616538/debug_22_4.tar.gz)

**Comment by leepc12** [comment]
An error occurred in peak-calling steps (SPP). We will look into this error.

**Comment by dcdxy** [comment]
Peak calling was set to MACS2; not sure if that makes a difference for this error.

**Comment by leepc12** [comment]
Please run this for debugging. It looks like you are using very old version of the pipeline.
```
$ cat /mnt/work1/users/home2/ddenisko/miniconda3/envs/encode-chip-seq-pipeline/bin/trimfastq.py
```

**Comment by dcdxy** [comment]
I pulled the repo earlier last week. Please find the output from that command below.

```
##################################                                                                                                                                                                                 
#                                #                                                                                                                                                                                 
# Last modified 2017/11/08       #                                                                                                                                                                                 
#                                #                                                                                                                                                                                 
# Georgi Marinov                 #                                                                                                                                                                                 
#                                #                                                                                                                                                                                 
##################################                                                                                                                                                                                 
                                                                                                                                                                                                                   
import sys                                                                                                                                                                                                         
import os                                                                                                                                                                                                          
                                                                                                                                                                                                                   
# try:                                                                                                                                                                                                             
#       import psyco                                                                                                                                                                                               
#       psyco.full()                                                                                                                                                                                               
# except:                                                                                                                                                                                                          
#       pass                                                                                                                                                                                                       
                                                                                                                                                                                                                   
def run():                                                                                                                                                                                                         
                                                                                                                                                                                                                   
    if len(sys.argv) < 2:                                                                                                                                                                                          
        print('usage: python %s <inputfilename> <bpToKeep | max> [-trim5 bp] [-flowcellID flowcell] [-addEnd 1 | 2] [-replace string newstring | blank] [-renameIDs prefix] [-stdout]' % sys.argv[0])              
        print('\tthe -trim5 option will trim additional bp from the 5 end, i.e. if you want the middle 36bp of 38bp reads, use 36 as bp to keep and 1 as the trim5 argument')                                      
        print('\tUse - to specify standard input, the script will print(to standard output by default')                                                                                                            
        print('\tThe script can read compressed files as long as they have the correct suffix - .bz2 or .gz')                                                                                                      
        sys.exit(1)                                                                                                                                                                                                
                                                                                                                                                                                                                   
    inputfilename = sys.argv[1]                                                                                                                                                                                    
    doMax=False                                                                                                                                                                                                    
    if sys.argv[2] == 'max':                                                                                                                                                                                       
        doMax=True                                                                                                                                                                                                 
        trim='max'                                                                                                                                                                                                 
    else:                                                                                                                                                                                                          
        trim = int(sys.argv[2])                                                                                                                                                                                    
    outputfilename = inputfilename.split('/')[-1].split('.fastq')[0] + '.' +str(trim)+'mers.fastq'                                                                                                                 
    doFlowcellID=False                                                                                                                                                                                             
                                                                                                                                                                                                                   
    doStdOut=True                                                                                                                                                                                                  
#    doStdOut=False                                                                                                                                                                                                
#    if '-stdout' in sys.argv:                                                                                                                                                                                     
#        doStdOut = True                                                                                                                                                                                           
                                                                                                                                                                                                                   
    if '-flowcellID' in sys.argv:                                                                                                                                                                                  
        doFlowcellID=True                                                                                                                                                                                          
        flowcellID=sys.argv[sys.argv.index('-flowcellID')+1]                                                                                                                                                       
        if doStdOut:                                                                                                                                                                                               
            pass                                                                                                                                                                                                   
        else:                                                                                                                                                                                                      
            print('will include flowcell ID', flowcellID, 'in reads headers')                                                                                                                                      
                                                                                                                                                                                                                   
    doRenameIDs = False                                                                                                                                                                                            
    if '-renameIDs' in sys.argv:                                                                                                                                                                                   
        doRenameIDs = True                                                                                                                                                                                         
        RID = '@' + sys.argv[sys.argv.index('-renameIDs') + 1]                                                                                                                                                     

    dotrim5=False                                                                                                                                                                                                  
    if '-trim5' in sys.argv:                                                                                                                                                                                       
        dotrim5=True                                                                                                                                                                                               
        trim5=int(sys.argv[sys.argv.index('-trim5')+1])                                                                                                                                                            
        if doStdOut:                                                                                                                                                                                               
            pass                                                                                                                                                                                                   
        else:                                                                                                                                                                                                      
            print('will trim ', trim5, 'bp from the 5-end')                                                                                                                                                        
        outputfilename = inputfilename.split('.fastq')[0] + '.' +str(trim)+'bp-5prim-trim.fastq'                                                                                                                   
                                                                                                                                                                                                                   
    doAddEnd=False                                                                                                                                                                                                 
    if '-addEnd' in sys.argv:                                                                                                                                                                                      
        doAddEnd=True                                                                                                                                                                                              
        END=sys.argv[sys.argv.index('-addEnd')+1]                                                                                                                                                                  
        if doStdOut:                                                                                                                                                                                               
            pass                                                                                                                                                                                                   
        else:                                                                                                                                                                                                      
            print('will add',  '/'+END, 'to read IDs')                                                                                                                                                             
                                                                                                                                                                                                                   
    doReplace=False                                                                                                                                                                                                
    if '-replace' in sys.argv:                                                                                                                                                                                     
        doReplace=True                                                                                                                                                                                             
        oldstring=sys.argv[sys.argv.index('-replace')+1]                                                                                                                                                           
        newstring=sys.argv[sys.argv.index('-replace')+2]                                                                                                                                                           
        if newstring == 'blank':                                                                                                                                                                                   
            newstring=''                                                                                                                                                                                           
        if doStdOut:                                                                                                                                                                                               
            pass                                                                                                                                                                                                   
        else:                                                                                                                                                                                                      
            print('will replace',  oldstring, 'with', newstring, 'in read IDs')                                                                                                                                    
                                                                                                                                                                                                                   
    i=0                                                                                                                                                                                                            
    shorter=0                                                                                                                                                                                                      
                                                                                                                                                                                                                   
    if doStdOut:                                                                                                                                                                                                   
        pass                                                                                                                                                                                                       
    else:                                                                                                                                                                                                          
        outfile = open(outputfilename, 'w')                                                                                                                                                                        
                                                                                                                                                                                                                   
    doStdIn = False                                                                                                                                                                                                
    if inputfilename != '-':                                                                                                                                                                                       
        if inputfilename.endswith('.bz2'):                                                                                                                                                                         
            cmd = 'bzip2 -cd ' + inputfilename                                                                                                                                                                     
        elif inputfilename.endswith('.gz'):                                                                                                                                                                        
            cmd = 'gunzip -c ' + inputfilename                                                                                                                                                                     
        else:                                                                                                                                                                                                      
            cmd = 'cat ' + inputfilename                                                                                                                                                                           
        p = os.popen(cmd, "r")                                                                                                                                                                                     
    else:                                                                                                                                                                                                          
        doStdIn = True                                                                                                                                                                                             
                                                                                                                                                                                                                   
    line = 'line'
                                                                                                                                                                                                        
    if dotrim5:                                                                                                                                                                                                    
        i=1
        j=0                                                                                                                                                                                                        
        while line != '':                                                                                                                                                                                          
            if doStdIn:                                                                                                                                                                                            
                line = sys.stdin.readline()                                                                                                                                                                        
            else:                                                                                                                                                                                                  
                line = p.readline()                                                                                                                                                                                
            if line == '':                                                                                                                                                                                         
                break                                                                                                                                                                                              
            if i==1 and line[0]=='@':                                                                                                                                                                              
                if doFlowcellID and flowcellID not in line:                                                                                                                                                        
                    ID='@'+flowcellID+'_'+line.replace(' ','_')[1:-1]+'\n'                                                                                                                                         
                else:                                                                                                                                                                                              
                    ID=line.replace(' ','_')                                                                                                                                                                       
                if doReplace:                                                                                                                                                                                      
                    ID=ID.replace(oldstring,newstring)                                                                                                                                                             
                if doRenameIDs:                                                                                                                                                                                    
                    ID = RID + str(j)                                                                                                                                                                              
                if doAddEnd:                                                                                                                                                                                       
                    ID=ID.strip()+'/'+END+'\n'                                                                                                                                                                     
                i=2                                                                                                                                                                                                
                continue                                                                                                                                                                                           
            if i==2:                                                                                                                                                                                               
                i=3                                                                                                                                                                                                
                sequence=line[trim5:len(line)].strip()                                                                                                                                                             
                continue                                                                                                                                                                                           
            if i==3 and line[0]=='+':                                                                                                                                                                              
                plus='+\n'                                                                                                                                                                                         
                i=4                                                                                                                                                                                                
                continue                                                                                                                                                                                           
            if i==4:                                                                                                                                                                                               
                scores=line                                                                                                                                                                                        
                i=1                                                                                                                                                                                                
                scores=line[trim5:len(line)].strip()                                                                                                                                                               
                scores=scores[0:trim]                                                                                                                                                                              
                j=j+1                                                                                                                                                                                              
                if j % 5000000 == 0:                                                                                                                                                                               
                    if doStdOut:                                                                                                                                                                                   
                        pass                                                                                                                                                                                       
                    else:                                                                                                                                                                                          
                        print(str(j/1000000) + 'M reads processed')                                                                                                                                                
                if doMax:                                                                                                                                                                                          
                    sequence=sequence.replace('.','N')                                                                                                                                                             
                else:                               
                    sequence=sequence[0:trim].replace('.','N')+'\n'
                if doStdOut:                        
                    print(ID.strip())               
                    print(sequence.strip())         
                    print(plus.strip())             
                    print(scores)                   
                else:                               
                    outfile.write(ID.strip()+'\n')  
                    outfile.write(sequence.strip()+'\n')
                    outfile.write(plus.strip()+'\n')
                    outfile.write(scores + '\n')    
                continue
    else:                                           
        i=1                                         
        j=0                                         
        while line != '':                           
            if doStdIn:                             
                line = sys.stdin.readline()         
            else:                                   
                line = p.readline()                 
            if line == '':                          
                break                               
            if i==1 and line[0]=='@':               
                if doFlowcellID and flowcellID not in line:
                    ID='@'+flowcellID+'_'+line.replace(' ','_')[1:-1]+'\n'
                else:                               
                    ID=line.replace(' ','_')        
                if doReplace:                       
                    ID=ID.replace(oldstring,newstring)
                if doRenameIDs:                     
                    ID = RID + str(j)
                if doAddEnd:                        
                    ID=ID.strip()+'/'+END+'\n'      
                i=2                                 
                continue                            
            if i==2:
                i=3                                 
                j=j+1                                                                                    
                if j % 5000000 == 0:                
                    if doStdOut:
                        pass                                                                             
                    else:                           
                        print(str(j/1000000) + 'M reads processed')
                if doMax:                                                                                
                    sequence=line                   
                else:
                    if len(line.strip())<trim:      
                        shorter+=1
                        sequence=line.strip().replace('.','N')+'\n'
                    else:                                                                                
                        sequence=line[0:trim].replace('.','N')+'\n'
                continue
            if i==3 and line[0]=='+':
                plus='+\n'
                i=4
                continue
            if i==4:
                i=1
                if doMax: 
                    scores=line
                    if doStdOut:
                        print(ID.strip())
                        print(sequence.strip())
                        print(plus.strip())
                        print(line.strip())
                    else:
                        outfile.write(ID)
                        outfile.write(sequence)
                        outfile.write(plus)
                        outfile.write(line)
                else:
                    if len(line.strip())<trim:
                        continue
                    scores=line[0:trim]+'\n'
                    if doStdOut:
                        print(ID.strip())
                        print(sequence.strip())
                        print(plus.strip())
                        print(scores.strip())
                    else:
                        outfile.write(ID)
                        outfile.write(sequence)
                        outfile.write(plus)
                        outfile.write(scores)
                continue

    if doStdOut:
        pass
    else:
        outfile.close()

    if shorter>0:
        print(shorter, 'sequences shorter than desired length')
run()
```

**Comment by dcdxy** [comment]
Do you mean `encode-chip-seq-pipeline`? Please find the output below.

```
$ source deactivate
$ source activate encode-chip-seq-pipeline
$ which trimfastq.py
~/miniconda3/envs/encode-chip-seq-pipeline/bin/trimfastq.py
$ cat $(which trimfastq.py)
##################################                                                                                                                                                                                 
#                                #                                                                                                                                                                                 
# Last modified 2017/11/08       #                                                                                                                                                                                 
#                                #                                                                                                                                                                                 
# Georgi Marinov                 #                                                                                                                                                                                 
#                                #                                                                                                                                                                                 
##################################                                                                                                                                                                                 
                                                                                                                                                                                                                   
import sys                                                                                                                                                                                                         
import os                                                                                                                                                                                                          
                                                                                                                                                                                                                   
# try:                                                                                                                                                                                                             
#       import psyco                                                                                                                                                                                               
#       psyco.full()                                                                                                                                                                                               
# except:                                                                                                                                                                                                          
#       pass                                                                                                                                                                                                       
                                                                                                                                                                                                                   
def run():                                                                                                                                                                                                         
                                                                                                                                                                                                                   
    if len(sys.argv) < 2:                                                                                                                                                                                          
        print('usage: python %s <inputfilename> <bpToKeep | max> [-trim5 bp] [-flowcellID flowcell] [-addEnd 1 | 2] [-replace string newstring | blank] [-renameIDs prefix] [-stdout]' % sys.argv[0])              
        print('\tthe -trim5 option will trim additional bp from the 5 end, i.e. if you want the middle 36bp of 38bp reads, use 36 as bp to keep and 1 as the trim5 argument')                                      
        print('\tUse - to specify standard input, the script will print(to standard output by default')                                                                                                            
        print('\tThe script can read compressed files as long as they have the correct suffix - .bz2 or .gz')                                                                                                      
        sys.exit(1)                                                                                                                                                                                                
                                                                                                                                                                                                                   
    inputfilename = sys.argv[1]                                                                                                                                                                                    
    doMax=False                                                                                                                                                                                                    
    if sys.argv[2] == 'max':                                                                                                                                                                                       
        doMax=True                                                                                                                                                                                                 
        trim='max'                                                                                                                                                                                                 
    else:                                                                                                                                                                                                          
        trim = int(sys.argv[2])                                                                                                                                                                                    
    outputfilename = inputfilename.split('/')[-1].split('.fastq')[0] + '.' +str(trim)+'mers.fastq'                                                                                                                 
    doFlowcellID=False

    doStdOut=True                                                                                                                                                                                                  
#    doStdOut=False                                                                                                                                                                                                
#    if '-stdout' in sys.argv:                                                                                                                                                                                     
#        doStdOut = True                                                                                                                                                                                           
                                                                                                                                                                                                                   
    if '-flowcellID' in sys.argv:                                                                                                                                                                                  
        doFlowcellID=True                                                                                                                                                                                          
        flowcellID=sys.argv[sys.argv.index('-flowcellID')+1]                                                                                                                                                       
        if doStdOut:                                                                                                                                                                                               
            pass                                                                                                                                                                                                   
        else:                                                                                                                                                                                                      
            print('will include flowcell ID', flowcellID, 'in reads headers')                                                                                                                                      
                                                                                                                                                                                                                   
    doRenameIDs = False                                                                                                                                                                                            
    if '-renameIDs' in sys.argv:                                                                                                                                                                                   
        doRenameIDs = True                                                                                                                                                                                         
        RID = '@' + sys.argv[sys.argv.index('-renameIDs') + 1]                                                                                                                                                     
                                                                                                                                                                                                                   
    dotrim5=False                                                                                                                                                                                                  
    if '-trim5' in sys.argv:                                                                                                                                                                                       
        dotrim5=True                                                                                                                                                                                               
        trim5=int(sys.argv[sys.argv.index('-trim5')+1])                                                                                                                                                            
        if doStdOut:                                                                                                                                                                                               
            pass                                                                                                                                                                                                   
        else:                                                                                                                                                                                                      
            print('will trim ', trim5, 'bp from the 5-end')                                                                                                                                                        
        outputfilename = inputfilename.split('.fastq')[0] + '.' +str(trim)+'bp-5prim-trim.fastq'                                                                                                                   
                                                                                                                                                                                                                   
    doAddEnd=False                                                                                                                                                                                                 
    if '-addEnd' in sys.argv:                                                                                                                                                                                      
        doAddEnd=True                                                                                                                                                                                              
        END=sys.argv[sys.argv.index('-addEnd')+1]                                                                                                                                                                  
        if doStdOut:                                                                                                                                                                                               
            pass                                                                                                                                                                                                   
        else:                                                                                                                                                                                                      
            print('will add',  '/'+END, 'to read IDs')                                                                                                                                                             
                                                                                                                                                                                                                   
    doReplace=False                                                                                                                                                                                                
    if '-replace' in sys.argv:                                                                                                                                                                                     
        doReplace=True                                                                                                                                                                                             
        oldstring=sys.argv[sys.argv.index('-replace')+1]                                                                                                                                                           
        newstring=sys.argv[sys.argv.index('-replace')+2]                                                                                                                                                           
        if newstring == 'blank':                                                                                                                                                                                   
            newstring=''                                                                                                                                                                                           
        if doStdOut:                                                                                                                                                                                               
            pass                                                                                                                                                                                                   
        else:                                                                                                                                                                                                      
            print('will replace',  oldstring, 'with', newstring, 'in read IDs')                                                                                                                                    
                                                                                                                                                                                                                   
    i=0                                                                                                                                                                                                            
    shorter=0

    if doStdOut:                                                                                                                                                                                                   
        pass                                                                                                                                                                                                       
    else:                                                                                                                                                                                                          
        outfile = open(outputfilename, 'w')                                                                                                                                                                        
                                                                                                                                                                                                                   
    doStdIn = False                                                                                                                                                                                                
    if inputfilename != '-':                                                                                                                                                                                       
        if inputfilename.endswith('.bz2'):                                                                                                                                                                         
            cmd = 'bzip2 -cd ' + inputfilename                                                                                                                                                                     
        elif inputfilename.endswith('.gz'):                                                                                                                                                                        
            cmd = 'gunzip -c ' + inputfilename                                                                                                                                                                     
        else:                                                                                                                                                                                                      
            cmd = 'cat ' + inputfilename                                                                                                                                                                           
        p = os.popen(cmd, "r")                                                                                                                                                                                     
    else:                                                                                                                                                                                                          
        doStdIn = True                                                                                                                                                                                             
                                                                                                                                                                                                                   
    line = 'line'                                                                                                                                                                                                  
                                                                                                                                                                                                                   
    if dotrim5:                                                                                                                                                                                                    
        i=1                                                                                                                                                                                                        
        j=0                                                                                                                                                                                                        
        while line != '':                                                                                                                                                                                          
            if doStdIn:                                                                                                                                                                                            
                line = sys.stdin.readline()                                                                                                                                                                        
            else:                                                                                                                                                                                                  
                line = p.readline()                                                                                                                                                                                
            if line == '':                                                                                                                                                                                         
                break                                                                                                                                                                                              
            if i==1 and line[0]=='@':                                                                                                                                                                              
                if doFlowcellID and flowcellID not in line:                                                                                                                                                        
                    ID='@'+flowcellID+'_'+line.replace(' ','_')[1:-1]+'\n'                                                                                                                                         
                else:                                                                                                                                                                                              
                    ID=line.replace(' ','_')                                                                                                                                                                       
                if doReplace:                                                                                                                                                                                      
                    ID=ID.replace(oldstring,newstring)                                                                                                                                                             
                if doRenameIDs:                                                                                                                                                                                    
                    ID = RID + str(j)                                                                                                                                                                              
                if doAddEnd:                                                                                                                                                                                       
                    ID=ID.strip()+'/'+END+'\n'                                                                                                                                                                     
                i=2                                                                                                                                                                                                
                continue                                                                                                                                                                                           
            if i==2:                                                                                                                                                                                               
                i=3                                                                                                                                                                                                
                sequence=line[trim5:len(line)].strip()                                                                                                                                                             
                continue                                                                                                                                                                                           
            if i==3 and line[0]=='+':                                                                                                                                                                              
                plus='+\n'                                                                                                                                                                                         
                i=4                                                                                                                                                                                                
                continue                                                                                                                                                                                           
            if i==4:                                                                                                                                                                                               
                scores=line
                i=1                                                                                                                     
                scores=line[trim5:len(line)].strip()                                                                                                                                                               
                scores=scores[0:trim]                                                                                                                                                                              
                j=j+1                                                                                                                                                                                              
                if j % 5000000 == 0:                                                                                                                                                                               
                    if doStdOut:                                                                                                                                                                                   
                        pass                                                                                                                                                                                       
                    else:                                                                                                                                                                                          
                        print(str(j/1000000) + 'M reads processed')                                                                                                                                                
                if doMax:                                                                                                                                                                                          
                    sequence=sequence.replace('.','N')                                                                                                                                                             
                else:                                                                                                                                                                                              
                    sequence=sequence[0:trim].replace('.','N')+'\n'                                                                                                                                                
                if doStdOut:                                                                                                                                                                                       
                    print(ID.strip())                                                                                                                                                                              
                    print(sequence.strip())                                                                                                                                                                        
                    print(plus.strip())                                                                                                                                                                            
                    print(scores)                                                                                                                                                                                  
                else:                                                                                                                                                                                              
                    outfile.write(ID.strip()+'\n')                                                                                                                                                                 
                    outfile.write(sequence.strip()+'\n')                                                                                                                                                           
                    outfile.write(plus.strip()+'\n')                                                                                                                                                               
                    outfile.write(scores + '\n')                                                                                                                                                                   
                continue                                                                                                                                                                                           
    else:                                                                                                                                                                                                          
        i=1                                                                                                                                                                                                        
        j=0                                                                                                                                                                                                        
        while line != '':                                                                                                                                                                                          
            if doStdIn:                                                                                                                                                                                            
                line = sys.stdin.readline()                                                                                                                                                                        
            else:                                                                                                                                                                                                  
                line = p.readline()                                                                                                                                                                                
            if line == '':                                                                                                                                                                                         
                break                                                                                                                                                                                              
            if i==1 and line[0]=='@':                                                                                                                                                                              
                if doFlowcellID and flowcellID not in line:                                                                                                                                                        
                    ID='@'+flowcellID+'_'+line.replace(' ','_')[1:-1]+'\n'                                                                                                                                         
                else:                                                                                                                                                                                              
                    ID=line.replace(' ','_')                                                                                                                                                                       
                if doReplace:                                                                                                                                                                                      
                    ID=ID.replace(oldstring,newstring)                                                                                                                                                             
                if doRenameIDs:                                                                                                                                                                                    
                    ID = RID + str(j)                                                                                                                                                                              
                if doAddEnd:                                                                                                                                                                                       
                    ID=ID.strip()+'/'+END+'\n'                                                                                                                                                                     
                i=2                                                                                                                                                                                                
                continue                                                                                                                                                                                           
            if i==2:                                                                                                                                                                                               
                i=3                                                                                                                                                                                                
                j=j+1                                                                                                                                                                                              
                if j % 5000000 == 0:                                                                                                                                                                               
                    if doStdOut:                                                                                                                                                                                   
                        pass                                                                                                                                                                                       
                    else:                                                                                                                                                                                          
                        print(str(j/1000000) + 'M reads processed')
                if doMax:                           
                    sequence=line                   
                else:                               
                    if len(line.strip())<trim:      
                        shorter+=1                  
                        sequence=line.strip().replace('.','N')+'\n'                                      
                    else:                           
                        sequence=line[0:trim].replace('.','N')+'\n'                                      
                continue                            
            if i==3 and line[0]=='+':               
                plus='+\n'                          
                i=4                                 
                continue                            
            if i==4:                                
                i=1                                 
                if doMax:                           
                    scores=line                     
                    if doStdOut:                    
                        print(ID.strip())           
                        print(sequence.strip())     
                        print(plus.strip())         
                        print(line.strip())         
                    else:                           
                        outfile.write(ID)           
                        outfile.write(sequence)     
                        outfile.write(plus)         
                        outfile.write(line)         
                else:                               
                    if len(line.strip())<trim:      
                        continue                    
                    scores=line[0:trim]+'\n'        
                    if doStdOut:                    
                        print(ID.strip())           
                        print(sequence.strip())     
                        print(plus.strip())         
                        print(scores.strip())       
                    else:                           
                        outfile.write(ID)           
                        outfile.write(sequence)     
                        outfile.write(plus)         
                        outfile.write(scores)       
                continue                            

    if doStdOut:                                    
        pass                                        
    else:                                           
        outfile.close()                             

    if shorter>0:                                   
        print(shorter, 'sequences shorter than desired length')                                          
run() 
```

**Comment by leepc12** [comment]
Sorry, yes it should be chip-seq.

It's weird. Your error log says that `()` is missing for `print` in `main()` of `trimfastq.py` (python3 fashion). But `()` is there in your `trimfastq.py`.

**Comment by leepc12** [comment]
The test sample in the tutorial is mapped with a full reference BWA indices, which means that pipeline will require some memory enough to load BWA indices on it.

Do you have enough memory on your system?
```
$ free -h
```
Did you following instruction on https://github.com/ENCODE-DCC/chip-seq-pipeline2/blob/master/docs/tutorial_local_conda.md?

**Comment by dcdxy** [comment]
```
$ free -h
              total        used        free      shared  buff/cache   available
Mem:           7.6G        4.4G        1.2G        332M        1.9G        2.4G
Swap:            0B          0B          0B
```
Does the error mean that I should have at least 3.1 G available on my system to run the pipeline?

Yes, I followed the local Conda instructions.

**Comment by leepc12** [comment]
Yes, you currently have 2.4G available, which is too small to run the test sample.

I think even 7.6G is not enough to run a pipeline for the test sample since it should be mapped to a full GRCh38 reference. I will think about reducing the size of the reference genome.

**Comment by dcdxy** [comment]
Also 2 suggestions for the template JSONs:

- neither should not contain `,` on the last line before `}` (i.e. `"chip.spp_disks" : "local-disk 100 HDD",` throws an error)
- `"chip.title"` and `"chip.description"` in `template_pe.json` should say "paired-end" (instead of "single-end")

**Comment by leepc12** [comment]
Thanks for your suggestions. They will be fixed in the next release.

Did you activate conda environment before running java command line?

```
$ which trimfastq.py
$ source activate encode-chip-seq-pipeline
$ which trimfastq.py
```

**Comment by dcdxy** [comment]
Yes, I activated it. Please see below:

```
$ source deactivate
$ which trimfastq.py
~/miniconda3/envs/encode-chip-seq-pipeline/bin/trimfastq.py
$ source activate encode-chip-seq-pipeline
$ which trimfastq.py
~/miniconda3/envs/encode-chip-seq-pipeline/bin/trimfastq.py
```

**Comment by leepc12** [comment]
How did you run pipelines? Did you `qsub` a shell script which has a java command line in it? If so you should activate the environment inside the shell script.

If you are using `sge` backend (`java -jar ... -Dbackend.default=sge ...`) instead then you need to activate it before running the java command line.

**Comment by dcdxy** [comment]
I did the latter (`java -jar...`) with `sge` backend and activated the environment beforehand:

```
$ source activate encode-chip-seq-pipeline
$ java -jar -Dconfig.file=backends/backend.conf -Dbackend.default=sge cromwell-34.jar run atac.wdl -i ${INPUT} -o workflow_opts/sge.json
```

**Comment by leepc12** [comment]
Can you send me one of your R1 FASTQs with 38 bp reads? My Dropbox account is leepc12 at gmail dot com.

**Comment by dcdxy** [comment]
Sure, I've shared the folder with you on Dropbox--please let me know if you have any trouble accessing its files.

**Comment by dcdxy** [comment]
Thanks very much for the fix! It looks like my pipeline runs were successful.
